include/cantera/thermo/CoverageDependentSurfPhase.h

```diff
@@ -0,0 +1,476 @@
+
+namespace Cantera
+{
+
+//! A thermodynamic model for a coverage-dependent surface phase, applying adsorbate
+//! lateral interaction correction factors to the ideal surface phase properties.
+/*!
+ * The ideal surface phase assumes no lateral interaction among adsorbates.
+ * This coverage-dependent surface phase allows adding lateral interaction
+ * correction terms to the ideal surface phase (SurfPhase) thermodynamic properties
+ * so that more accurate adsorbate thermochemistry can be achieved.
+ *
+ * ## Coverage-dependent Thermodynamic Properties Formulations
+ *
+ * At a low-coverage limit, an adsorbate thermochemistry is the same as
+ * that of ideal species since there are no adsorbates in the vicinity to cause
+ * lateral interaction. Therefore, it is logical to set ideal species properties
+ * as the low-coverage limit properties and add lateral interaction terms, i.e.
+ * lateral interaction correction terms, to them as excess properties.
+ * Accordingly, standard state coverage-dependent enthalpy, entropy,
+ * and heat capacity of a species \f$ k \f$ can be formulated as follows.
+ *
+ * \f[
+ *  h_k^o(T,\theta)
+ *      = \underbrace{h_k^{o,ideal}(T)
+ *        + \int_{298}^{T}cp_k^{o,ideal}(T)dT}_{\text{low-coverage limit}}
+ *        + \underbrace{h_k^{o,cov}(T,\theta)
+ *        + \int_{298}^{T}cp_k^{o,cov}(T,\theta)dT}_{\text{coverage dependence}}
+ *
+ * \f]
+ *
+ * \f[
+ *  s_k^o(T,\theta)
+ *     = \underbrace{s_k^{o,ideal}(T)
+ *       + \int_{298}^{T}\frac{cp_k^{o,ideal}(T)}{T}dT}_{\text{low-coverage limit}}
+ *       + \underbrace{s_k^{o,cov}(T,\theta)
+ *       + \int_{298}^{T}\frac{cp_k^{o,cov}(T,\theta)}{T}dT}_{\text{coverage
+ *         dependence}}
+ * \f]
+ *
+ * \f[
+ *  cp_k^o(T,\theta)
+ *      = cp_k^{o,ideal}(T) + cp_k^{o,cov}(T,\theta)
+ * \f]
+ *
+ * ## Mathematical Models for Coverage-dependent Correction Terms
+ *
+ * Coverage-dependent correction terms for enthalpy and entropy can be calculated
+ * with one of the four algebraic models: linear dependecy model, piecewise-linear
+ * dependency model, polynomial dependency model, and interpolative dependency model.
+ * In the dependency model equations, a coverage-dependent correction term is denoted
+ * by \f$ f^{cov} \f$ where \f$ f \f$ can be either enthalpy or entropy. Because
+ * lateral interaction can compose of both self- and cross- interactions, the total
+ * correction term of species \f$ k \f$ is a sum of all interacting species \f$ j \f$
+ * which can include itself. Coefficients \f$ c1_{k,j}-c7_{k,j} \f$ are user-provided
+ * parameters that can be given in input mechanism.
+ *
+ * Linear dependency model:
+ * \f[
+ *  f^{cov}_k(\theta) = \sum_j c1_{k,j} \theta_j
+ * \f]
+ *
+ * Piecewise-linear dependency model:
+ * \f[
+ * f^{cov}_k(\theta) = \sum_j \left\{
+ *  \begin{array}{ll}
+ *  c2_{k,j}\theta_j & \text{, } \theta_j \leq \theta^\text{change}_{k,j} \\
+ *  \left[c3_{k,j}(\theta_j - \theta^\text{change}_{k,j})
```

Comment:
The notation for the model coefficients $c2_{k,j}$ and the like are not very attractive when rendered as equations in the Doxygen documentation. I'm not sure what the best alternative. Maybe $c^{(2)}_{k,j}$?